tests/ecosystem/deployment/test_deployment.py

```diff
@@ -15,9 +16,139 @@ def test_deployment():
+    assert is_cluster_running(config.ENV_DATA['cluster_path'])
+
+    # Verify OCS Operator and Local Storage Operator in Succeeded phase
+    csv = ocp.OCP(kind='csv', namespace=namespace)
+    csvs = csv.get()
+    for item in csvs['items']:
+        name = item['metadata']['name']
+        log.info("Checking status of %s", name)
+        assert item['status']['phase'] == 'Succeeded', (
+            f"Operator {name} not 'Succeeded'"
+        )
+
+    # Verify OCS Cluster Service (ocs-storagecluster) is Ready
+    storage_cluster = ocp.OCP(kind='StorageCluster', namespace=namespace)
+    storage_clusters = storage_cluster.get()
+    for item in storage_clusters['items']:
+        name = item['metadata']['name']
+        log.info("Checking status of %s", name)
+        assert item['status']['phase'] == 'Ready', (
+            f"StorageCluster {name} not 'Ready'"
+        )
+
+    # Verify pods in running state and proper counts
+    pod = ocp.OCP(
+        kind=constants.POD, namespace=namespace
+    )
+    timeout = 0
+    # ocs-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='name=ocs-operator',
+        timeout=timeout
+    )
+    # storagecluster-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-operator',
+        timeout=timeout
+    )
+    # nooba
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=noobaa',
+        resource_count=2,
+        timeout=timeout
+    )
+    # local-storage-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='name=local-storage-operator',
+        timeout=timeout
+    )
+    # mons
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-mon',
+        resource_count=3,
+        timeout=timeout
+    )
+    # csi-cephfsplugin
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-cephfsplugin',
+        resource_count=3
+    )
+    # csi-cephfsplugin-provisioner
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-cephfsplugin-provisioner',
+        resource_count=2,
+        timeout=timeout
+    )
+    # csi-rbdplugin
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-rbdplugin',
+        resource_count=3,
+        timeout=timeout
+    )
+    # csi-rbdplugin-profisioner
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-rbdplugin-provisioner',
+        resource_count=2,
+        timeout=timeout
+    )
+    # osds
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-osd',
+        resource_count=3,
```

Comment:
is this osd pod count ? if Yes, is it fixed to 3 all time ?

Reply:
This is OSD pod count yes. I based all resource_counts on the [document](https://docs.google.com/document/d/1iQfIGR1N_Uezzo0gs-pSDFQ1q_oANWZGGIuslhmdQqg/edit) linked in the jira ticket